sdk/opendp/whitenoise/client/__init__.py

```diff
@@ -34,24 +52,71 @@ def submit(self, params, uri):
         return client.executerun(details)
 
 class DatasetClient(object):
-    def register(self, dataset):
+    """
+    A client for registering, reading and releasing differentially private
+    datasets using the opendp-whitenoise service
+    """
+    def __init__(self):
+        # Tag requests with this custom header for now
+        self._guid = _guid_header()
+
+    def readreleased(self, dataset_name, budget):
+        """
+        Generates a DatasetReadReleaseRequest and sends it to the service.
+        Tags the request with Client guid.
+        """
         client = _get_client()
-
+        read_released_request = DatasetReadReleaseRequest(dataset_name=dataset_name, budget=budget)
+        return client.datasetreadreleased(read_released_request, custom_headers={'client_guid': self._guid})
+
+    def _register_release_request_helper(self, dataset):
+        """
+        Helper for register/release, 
+        both of which use DatasetDocuments as request formats
+        """
         for key in KNOWN_DATASET_TYPE_KEYS:
             if not key in dataset:
                 dataset[key]=None
 
-        register_request = DatasetDocument(dataset_name=dataset['dataset_name'], \
+        request = DatasetDocument(dataset_name=dataset['dataset_name'], \
             dataset_type=dataset['dataset_type'], \
             budget=dataset['budget'], \
             csv_details=dataset['csv_details'], \
-            dataverse_details=dataset['dataverse_details'])
-        return client.datasetregister(register_request)
+            dataverse_details=dataset['dataverse_details'], \
+            authorized_users=dataset['authorized_users'])
+        
+        return request
+
+    def release(self, dataset):
+        """
+        Generates a DatasetDocument and sends it to the service.
+        Requests the release of a Differentially Private DatasetDocument, with budget
+        (to authorized users)
+        Tags the request with Client guid.
+        """
+        client = _get_client()
+        release_request = self._register_release_request_helper(dataset)
+        return client.datasetrelease(release_request, custom_headers={'client_guid': self._guid})
 
-    def read(self, dataset_name, budget):
+    def register(self, dataset):
+        """
+        Generates a DatasetDocument and sends it to the service.
+        Requests the registration of this private DatasetDocument
+        Tags the request with Client guid.
+        """
+        client = _get_client()
+        register_request = self._register_release_request_helper(dataset)
+        return client.datasetregister(register_request, custom_headers={'client_guid': self._guid})
```

Comment:
custom_headers=self._custom_headers might be worth it. And you can add that in the constructor.

I did not expect you to add the guid already :) pleasant surprise